06.wc/wc.rb

```diff
@@ -44,43 +42,34 @@ def count_sentence(sentence, file_name = nil)
   }
 end
 
-def output(file_details, options)
-  total_value = calc_total(file_details)
-  col_width = calc_col_width(file_details, total_value)
-  file_details.each do |file_detail|
-    print_detail_value(file_detail, col_width, options)
-    print "#{file_detail[:name]}\n"
+def output_lines(text_details, options)
+  total_value = calc_total(text_details)
+  col_width = total_value.values_at(:line, :word, :byte).max.to_s.length
+  text_details.each do |text_detail|
+    output_line(text_detail, col_width, options)
   end
-  return if file_details.size == 1
+  return if text_details.size == 1
 
-  print_detail_value(total_value, col_width, options)
-  print "total\n"
+  output_line(total_value, col_width, options)
 end
 
-def calc_total(file_details)
-  total_line = 0
-  total_word = 0
-  total_byte = 0
-  file_details.each do |file_detail|
-    total_line += file_detail[:line]
-    total_word += file_detail[:word]
-    total_byte += file_detail[:byte]
+def calc_total(text_details)
+  total = Hash.new { |h, k| h[k] = 0 }
+  text_details.each do |text_detail|
+    total[:line] += text_detail[:line]
+    total[:word] += text_detail[:word]
+    total[:byte] += text_detail[:byte]
   end
-  {
-    line: total_line,
-    word: total_word,
-    byte: total_byte
-  }
-end
-
-def calc_col_width(file_details, total_value)
-  file_details[0][:name].nil? ? STDIN_COL_WIDTH : total_value.values.max.to_s.length
+  total[:name] = 'total'
+  total
 end
 
-def print_detail_value(detail_value, col_width, options)
+def output_line(detail_value, col_width, options)
   detail_value.each do |key, value|
```

Comment:
たぶんこのループではline, word、byte、nameの4つがループすると思いますが、nameはそもそもこの処理の出力対象ではないので

```suggestion
  %w(line word byte).each do |key|
```

のように出力対象のキーを明示的に書く方がベターだ（コードの読み手が首をかしげずに済む）と思います。

あと、Rubyの場合はハッシュにキーを追加した順番が保持されますが、他の言語だと順番を保証しないものも多いです。
なので、「ハッシュのキーが追加された順番に依存するプログラム」は極力避けた方がいいんじゃないかなーと思います（僕もたまに書いてしまいますが😅）。

Reply:
ご指摘に納得し%記法にしましたが、今回はシンボルのため、以下のように`%i`を利用した方が適しているのでは？と考えました。
```
%i[line word byte].each do |key|
```
%wの方がより良い、などの理由がありましたらご教示ください。

Reply:
あ、%wは深く考えずに書いてしまっただけです。すいません。
元のデータ型がシンボルなのであれば%iで大丈夫です！